src/main/java/org/kairosdb/datastore/cassandra/CassandraDatastore.java

```diff
@@ -413,6 +432,13 @@ private void queryWithRowKeys(DatastoreMetricQuery query,
             queryKeys.addAll(rowKeys);
         } else {
             for (DataPointsRowKey rowKey : rowKeys) {
+                long rowWidth = 0L;
+                //Use new row keys width based on the timestamp
+                if (rowKey.getTimestamp() >= m_newRowKeyStartTime) {
```

Comment:
It seems like we are copying this part, I suggest we replace it with a `calculateRowWidth()` or something.